src/lib/y2_alternatives/client/start_checking_permissions.rb

```diff
@@ -0,0 +1,32 @@
+# Copyright (c) 2016 SUSE LLC.
+#  All Rights Reserved.
+
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of version 2 or 3 of the GNU General
+#  Public License as published by the Free Software Foundation.
+
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+#  GNU General Public License for more details.
+
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, contact SUSE LLC.
+
+#  To contact SUSE about this file by physical or electronic mail,
+#  you may find current contact information at www.suse.com
+
+require "y2_alternatives/dialog/list_alternatives"
+
+Yast.import "Confirm"
+
+module Y2Alternatives
+  module Client
+    # Checks if user is root and create a ListAlternatives dialog
+    class StartCheckingPermissions
+      def main
+        Y2Alternatives::Dialog::ListAlternatives.run if Yast::Confirm.MustBeRoot
```

Comment:
in fact first `Y2Alternatives::` is not needed as code is already in this namespace, but it is just style taste. No strong opinion. so just note.